src/DiagnosticsUtil.cpp

```
@@ -0,0 +1,129 @@
+        snprintf(dst, dst_len, "****");
+        return;
+    }
+    // first 4 + "..." + last 4
+    snprintf(dst, dst_len, "%.4s...%s", uid, uid + (n - 4));
+}
+
+// Emit only scheme://host[:port] — dropping userinfo (user:pass@), the path,
+// and the query string, since any of those can carry a credential
+// (?api_key=..., /token/..., user:pass@). Reachability triage only needs the
+// host and port.
+void diagRedactUrl(char* dst, size_t dst_len, const char* url) {
```

Cppcheck (2.21.0)

[style] The function 'diagRedactUrl' is never used.

(unusedFunction)